File: sdk/ios/Sources/OpenStorySDK/Internal/Context/UserSegmentsNormalizer.swift
import Foundation

internal enum UserSegmentsNormalizer {
    static func normalize(_ rawSegments: some Collection<String>) -> [String] {
        Array(
            Set(
                rawSegments
                    .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
                    .filter { !$0.isEmpty }
            )
        )
        .sorted()
    }
}

File: sdk/flutter/android/src/main/kotlin/com/openstory/sdk/internal/context/UserSegmentsNormalizer.kt
package com.openstory.sdk.internal.context

internal object UserSegmentsNormalizer {
    fun normalize(rawSegments: Collection<String>): List<String> {
        return rawSegments
            .map { it.trim() }
            .filter { it.isNotEmpty() }
            .distinct()
            .sorted()
    }
}
